### PR TITLE
fix(components): [dialog] fix type error

### PR DESCRIPTION
**Pull request by Chuck-Lau**
#6114 

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/dialog/src/use-dialog.ts [hunk 1]
```diff
@@ -45,13 +44,13 @@ export const useDialog = (
 
   const style = computed<CSSProperties>(() => {
     const style: CSSProperties = {}
-    const varPrefix = `--${namespace.value}-dialog`
+    const varPrefix = `${namespace.value}-dialog`
```
Comment (holazz) [inline packages/components/dialog/src/use-dialog.ts:47]: I think it's better to use the const assertion directly here.

```ts
const varPrefix = `--${namespace.value}-dialog` as const
```

Comment (Tsong-LC) [reply packages/components/dialog/src/use-dialog.ts:47]: What about using `module enhancements`?

Comment (Tsong-LC) [reply packages/components/dialog/src/use-dialog.ts:47]: image

Comment (holazz) [reply packages/components/dialog/src/use-dialog.ts:47]: > I think it's better to use the const assertion directly here.
> 
> ```ts
> const varPrefix = `--${namespace.value}-dialog` as const
> ```

The const assertion here is to narrow the type `string` to ``` `--${string}` ```, what do you mean?

Comment (Tsong-LC) [reply packages/components/dialog/src/use-dialog.ts]: Updated. Thanks for your advice

**Review by ryuhangyeong — APPROVED** [review]

**Review by tolking — APPROVED** [review]

**Pull request merged**